operator/api/v1alpha1/odigos_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/odigos-io/odigos/common"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OdigosSpec defines the desired state of Odigos
+type OdigosSpec struct {
```

Comment:
we can consider adding the `mount-method` here as well:

https://docs.odigos.io/instrumentations/configuration/mount-method#mount-method

so users can change the instrumentation device if they observe issues with this method. it was added lately to the config as well